corehq/apps/reports/templates/reports/form/partials/single_form.html

```diff
@@ -51,11 +51,16 @@
         $("#archive-form").submit(function() {
             document.getElementById('archive-form-btn').disabled=true;
             document.getElementById('archive-spinner').style.display='inline-block';
-            ga_track_event('Reports', 'Case History', 'Archive this form', {
-                'hitCallback': function () {
+
+            // _.after(2,...) means the callback will only be called after *both* analytics
+            // functions have finished.
+            var analyticsCallback = _.after(2, function() {
```

Comment:
Hm.. I haven't actually tested that. But, I'm pretty sure that both the Goggle Analytics and KISSmetrics functions call the callback after a short timeout if the event hasn't been successfully tracked by then incase the the server is down or being slow or whatever.